packages/amplify-provider-awscloudformation/src/admin-modelgen.ts

```diff
@@ -25,88 +23,99 @@ export const adminModelgen = async (context: $TSContext, resources: $TSAny[]): P
   const { resourceName } = appSyncResource;
 
   const amplifyMeta = stateManager.getMeta();
-  const localEnvInfo = stateManager.getLocalEnvInfo();
 
   const appId = amplifyMeta?.providers?.[providerName]?.AmplifyAppId;
 
   if (!appId) {
     return;
   }
 
-  const { envName } = localEnvInfo;
   const { isAdminApp } = await isAmplifyAdminApp(appId);
   const isDSEnabled = await isDataStoreEnabled(path.join(pathManager.getBackendDirPath(), 'api', resourceName));
 
   if (!isAdminApp || !isDSEnabled) {
     return;
   }
 
-  const spinner = ora('Generating models in the cloud...\n').start();
-  const amplifyBackendInstance = await AmplifyBackend.getInstance(context);
+  // the following is a hack to enable us to upload assets needed by studio CMS to the deployment bucket without
+  // calling AmplifyBackend.generateBackendAPIModels.
+  // Calling this API introduces a circular dependency because this API in turn executes the CLI to generate codegen assets
 
+  const originalProjectConfig = stateManager.getProjectConfig();
+  const relativeTempOutputDir = 'amplify-codegen-temp';
+  const absoluteTempOutputDir = path.join(pathManager.findProjectRoot(), relativeTempOutputDir);
+  const forceJSCodegenProjectConfig = {
+    projectName: 'tempCodegen',
+    version: '3.1',
```

Comment:
is this a duplicate of a codegen configuration? does it need to be in sync? if so are there any mechanism we can put in place or at least a comment

Reply:
this isn't needed at all; removed to avoid confusion